blog/_posts/2021-10-19-scala-3.1.0-released.md

```diff
@@ -0,0 +1,168 @@
+---
+layout: blog-detail
+post-type: blog
+by: Paweł Marks, VirtusLab
+title: Scala 3.1.0 released!
+---
+Hello from the Scala 3 team! It has already been six weeks since we have announced the release candidate for the first minor version after the initial release of Scala 3. Now, after two more RCs, we can confidently promote Scala 3.1.0-RC3 to a new stable release of the language. In other words, Scala 3.1.0 is officially out!
+
+## Compatibility notcie
+
+This is a first *minor* release after the initial release of Scala 3.0. This has following consequences:
+
+- Scala 3.1 is backward source compatible: any code that was working in 3.0 will also work in 3.1.
+- Scala 3.1 is backward binary compatible: you can use dependencies compiled with Scala 3.0 in 3.1 projects.
+- Scala 3.1 is _not_ forward binary compatible: you _cannot_ use dependencies compiled with Scala 3.1 in Scala 3.0 projects.
+- Updating Scala to 3.1 is not forward compatible change. If you are a library maintainer you should only do that in minor release of your project.
+
+## What's new in 3.1
+
+### New experimental feature: safer exceptions
+
+A new experimental feature that allows declaring and checking which exceptions can be thrown. It relies on the effects as implicit capabilities pattern.
+
+You can opt-in using following import.
+
+```scala
+import language.experimental.saferExceptions
+```
+
+Now it is possible to mark types with exception that can be thrown during the evaluation:
+
+```scala
+def f(x: Double): Double canThrow LimitExceeded =
+  if x < limit then f(x) else throw LimitExceeded()
+```
+
+You can read more in [the document proposing this feature](https://github.com/lampepfl/dotty/blob/release-3.1.0/docs/docs/reference/experimental/canthrow.md).
+
+### Efficient bytecode for matching over strings
```

Comment:
would be very great to have a section for hierarchical mirror generation: https://github.com/lampepfl/dotty/pull/11686 e.g. 

### Summon Mirrors for Hierarchical Sealed Types
Consider the following sealed hierarchy:
```scala
sealed trait Top
case class Middle() extends Top with Bottom
sealed trait Bottom extends Top
```
Previously you would not be able to summon an instance of
`scala.deriving.Mirror.SumOf` for `Top`, as its child `Bottom`
is also sealed. In Scala 3.1 this is now possible, enabling
compiletime reflection over nested hierarchies of sealed types.

The only change necessary for users is that you should now
consider that one of the `MirroredElemTypes` for a `Mirror.SumOf`
could also be a sum type, not just product types.